lib/rspec/mocks/exclude_stubbed_classes_from_subclasses.rb

```diff
@@ -0,0 +1,53 @@
+module RSpec
+  module Mocks
+    # Support for `exclude_stubbed_classes_from_subclasses` configuration.
+    #
+    # @private
+    class ExcludeStubbedClassesFromSubclasses
+      def self.enable!
+        return unless RUBY_VERSION >= "3.1"
+        return if Class.respond_to?(:subclasses_with_rspec_mocks)
+
+        require 'weakref'
+
+        mod_something = Module.new do
+          def excluded_subclasses
+            @excluded_subclasses ||= []
+            @excluded_subclasses.select(&:weakref_alive?).map do |ref|
+              begin
+                ref.__getobj__
+              rescue RefError
+                nil
+              end
+            end.compact
+          end
+
+          def exclude_subclass(constant)
+            @excluded_subclasses ||= []
+            @excluded_subclasses << WeakRef.new(constant)
+          end
+        end
+        RSpec::Mocks.extend(mod_something)
+
+        Class.class_eval do
+          def subclasses_with_rspec_mocks
+            subclasses_without_rspec_mocks - RSpec::Mocks.excluded_subclasses
+          end
+
+          alias subclasses_without_rspec_mocks subclasses
+          alias subclasses subclasses_with_rspec_mocks
+        end
+      end
+
+      def self.disable!
+        return unless Class.respond_to?(:subclasses_with_rspec_mocks)
+
+        Class.class_eval do
+          undef subclasses_with_rspec_mocks
+          alias subclasses subclasses_without_rspec_mocks
```

Comment:
Rubocop triggers this warning : 

```
lib/rspec/mocks/exclude_stubbed_classes_from_subclasses.rb:47:11: W: Lint/DuplicateMethods: Method RSpec::Mocks::ExcludeStubbedClassesFromSubclasses::Class#subclasses is defined at both lib/rspec/mocks/exclude_stubbed_classes_from_subclasses.rb:38 and lib/rspec/mocks/exclude_stubbed_classes_from_subclasses.rb:47.
          alias subclasses subclasses_without_rspec_mocks
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

It's wrong because the method is undefined at line 46. I disabled this warning.